musicround/helpers/import_helper.py

```
@@ -233,46 +392,106 @@ def import_spotify_track(sp, track_id):
-                # No ISRC available, just use Spotify data
-                song = ImportHelper._create_song_from_spotify(track_info)
+                    current_app.logger.info(f"No comprehensive metadata found for ISRC {isrc}, or metadata was incomplete. Falling back for track {track_id}.")
+            
+            if song is None: # If no ISRC, or ISRC lookup failed to produce a song object
+                current_app.logger.info(f"Creating song for track {track_id} using basic Spotify data (no ISRC or failed ISRC enrichment).")
+                song = ImportHelper._create_song_from_spotify(track_info) # Ensure this helper is robust
             
-            # We already checked for duplicates above, so we can add the song directly
             if song:
+                # Ensure spotify_id is set if created via ISRC path primarily
+                if not song.spotify_id: song.spotify_id = track_id
```

Ruff (0.11.9)

Multiple statements on one line (colon)

(E701)